src/components/Channel/Channel.tsx

```diff
@@ -418,14 +418,11 @@ const ChannelInner = <
   }, [channel.cid, doMarkReadRequest]);
 
   useEffect(() => {
-    if (state.thread && state.messages?.length) {
-      for (let i = state.messages.length - 1; i >= 0; i -= 1) {
```

Comment:
This seems like a micro-optimization to search the messages backwards. I would rather swap that for readability.